### PR TITLE
Check stdout even when exit_status is 0, and handle JSON.parse exception

### PR DESCRIPTION
**Pull request by CodesWhisperer**
## Description
When the target is a cisco_ios device, the exit_status is 0 but the error is described in the stdout, so the if condition doesn't work and the JSON.parse raises an exception.

Also, the JSON.parse should always be in rescue block.

## Related Issue
https://github.com/inspec/train/issues/599

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by clintoncwolfe** [comment]
Code looks reasonable to me, just need to get it past the linter. Also please sign your commit to indicate acceptance of our contribution terms - https://github.com/inspec/train/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco

**Comment by CodesWhisperer** [comment]
@clintoncwolfe 
I fixed it, but the tests "tests-ruby-2.6-windows" didn't pass and I don't know why.

**Comment by clintoncwolfe** [comment]
The gem ffi segfaults under windows right now. We have a pending PR to pin the version back to a working version. Once that merges you can rebase to master and the windows tests should pass.

File: lib/train/platforms/detect/helpers/os_windows.rb
Comment (james-stocks) [inline lib/train/platforms/detect/helpers/os_windows.rb:36]: `!(command.stdout.downcase =~ /window/)` could be slightly simpler `command.stdout.downcase !~ /window/`

This new condition seems to make `command.stdout.empty?` redundant so the line could be

`return false if (command.exit_status != 0)  || command.stdout.downcase !~ /window/`

It may be valuable to leave a comment line above this statement to explain this case (some targets e.g. Cisco may return 0 and print an error to stdout)

**Comment by james-stocks** [comment]
Hi @CodesWhisperer -

This looks good, thank you! I left one comment suggesting to make the code easier to understand; because in future an issue may occur with some other platform; and the logic might get changed in a way that fixes that other platform but causes Cisco to start failing again...

As Clinton said, please rebase this branch against master to get the Windows tests working.

Thanks

**Comment by tas50** [comment]
@CodesWhisperer Can you rebase this against master please?

**Comment by rmoles** [comment]
It would be good to get this merged. Could you rebase against master @CodesWhisperer ?

**Comment by rmoles** [comment]
@CodesWhisperer unfortunately your last commit is missing your DCO, could you please push another commit with your DCO added?

**Comment by CodesWhisperer** [comment]
@rmoles Is it good?

**Comment by rmoles** [comment]
Awesome thanks @CodesWhisperer. Unfortunately, I don't have merge rights on this repo. Perhaps, @james-stocks or @Schwad can merge this.

**Review by james-stocks — APPROVED** [review]

**Comment by james-stocks** [comment]
Thanks for this @CodesWhisperer !

**Pull request merged**
